pkg/signup/service/signup_service.go

```diff
@@ -572,3 +621,30 @@ func getAppsURL(appRouteName string, signup signup.Signup) string {
 	appsURL := signup.ConsoleURL[index:]
 	return fmt.Sprintf("https://%s%s", appRouteName, appsURL)
 }
+
+func pollUpdateSignup(ctx *gin.Context, updater func() error) error {
+	// Attempt to execute an update function, retrying a number of times if the update fails
+	attempts := 0
+	for {
+		attempts++
```

Comment:
i guess the only times when a request will fail will be either when there is a network connectivity issue (unlikely since registration service is running within the same cluster) or when the UserSignup is somehow updated elsewhere between registration service loading it and attempting the update.  In the second case, a wait() won't really help here as a simple reload and re-attempt to update will be sufficient without any delays required.

Reply:
OK. Good point.